include/lbfgs.h

```diff
@@ -26,8 +26,8 @@
 
 /* $Id$ */
 
-#ifndef __LBFGS_H__
-#define __LBFGS_H__
+#ifndef LBFGS_H_
+#define LBFGS_H_
```

Comment:
It's no trouble, I've already done it :)

It's not specific to LLVM, it's the C language standard that specifies these limits on underscores.

And since these warning are in headers, it makes it harder for projects that include your headers to turn this warning on. (Which was my motivation for this patch.)